src/main/java/com/google/devtools/build/lib/rules/cpp/CcLinkingHelper.java

```diff
@@ -178,6 +178,14 @@ public CcLinkingHelper fromCommon(RuleContext ruleContext, CcCommon common) {
     return this;
   }
 
+  /** Sets fields that only exist in cc_library rules. */
+  public CcLinkingHelper fromLibrary(RuleContext ruleContext) {
```

Comment:
Please remove this method and just pass every CcLinkingContext from CcLibrary.java as usual.